pkg/k8s/watchers/cilium_endpoint.go

```diff
@@ -235,3 +252,56 @@ func (k *K8sWatcher) endpointDeleted(endpoint *types.CiliumEndpoint) {
 		k.egressGatewayManager.OnDeleteEndpoint(endpoint)
 	}
 }
+
+// ciliumEndpointManager manages tasks related to ciliumendpoints, including
+// performing periodic cleanup of local stale CEP resources.
+type ciliumEndpointManager struct {
+	*lock.RWMutex
+	ciliumEndpointsCleanupEnabled bool
+	markedForGC                   []*types.CiliumEndpoint
+	client                        v2.CiliumV2Interface
+}
+
+func newCiliumEndpointManager(ciliumClient v2.CiliumV2Interface) *ciliumEndpointManager {
+	return &ciliumEndpointManager{
+		RWMutex: &lock.RWMutex{},
+		client:  ciliumClient,
+	}
+}
+
+func (cpm *ciliumEndpointManager) markForDeletion(cep *types.CiliumEndpoint) {
+	cpm.Lock()
+	defer cpm.Unlock()
+	cpm.markedForGC = append(cpm.markedForGC, cep)
+}
+
+func (cpm *ciliumEndpointManager) sweep(ctx context.Context) error {
+	cpm.Lock()
+	defer cpm.Unlock()
+	for i := len(cpm.markedForGC) - 1; i >= 0; i-- {
+		cep := cpm.markedForGC[i]
+		if err := cpm.client.CiliumEndpoints(cep.Namespace).Delete(ctx, cep.Name, metav1.DeleteOptions{}); err != nil {
```

Comment:
We might be doing unnecessary API calls if the CEP is already GCed by Cilium Operator. We should check if the CEP exists in the local store and only if it doesn't then delete the object.

Reply:
Yeah good point, there's still the chance that the object has already been deleted and not processed but that would catch most cases.